src/twisted/internet/defer.py

```diff
@@ -91,7 +91,7 @@ def logError(err):
     return err
 
 
-def succeed(result):
+def succeed(result: object) -> "Deferred":
```

Comment:
For these particular functions where you added type annotations, do you want to keep  the  @rtype's in the docstrings, or do you want to get rid of them?

Reply:
I meant to get rid of them. Perhaps I forgot to do a final scan… yeah I totally skipped that for `interfaces.py`… and in doing so found a better type hint for one thing….

But this is `defer.py`… a bit harder to catch the ancillary type additions.  Fixing.

Reply:
Worth noting here that this type annotation will need to be updated when we make Deferred a generic.  I'd almost rather we do that *before* adding types all over the place in this module, since this gives the illusion of type safety without actually ensuring anything about the result type in the Deferred.